File: Subarray_With_Given_Sum.cpp
    vector<int> subarraySum(vector<int>arr, int n, long long s)
    {
        // Your code here
       if(s==0)return {-1};
               
       int i=0,j=0,sum=0;
       while(j<=n){
           if(sum<s){
               sum+=arr[j];
               j++;
           }
           else if(sum==s){
               return {i+1,j}; 
           }
           else if(sum>s){
               sum-=arr[i];
               i++;
           }
       }
       return {-1};
    } 

File: Subarray_With_Given_Sum.java
class Solution
{
    //Function to find a continuous sub-array which adds up to a given number.
    static ArrayList<Integer> subarraySum(int[] arr, int n, int s) 
    {
        // Your code here
        int sm=0;
        int l=0;
        int r=0;
        if(s==0)
        {
            ArrayList<Integer> ans=new ArrayList<>();
            ans.add(-1);
            return ans;
        }
        while(r<n)
        {
            if(sm<s)
            {
                sm+=arr[r];
            }
            if(sm>s)
            {
                while(sm>s)
                {
                    sm-=arr[l];
                    l+=1;
                }
            }
            if(sm==s)
            {
                ArrayList<Integer> ans=new ArrayList<>();
                ans.add(l+1);
                ans.add(r+1);
                return ans;
            }
            r +=1;
        }
        ArrayList<Integer> ans=new ArrayList<>();
        ans.add(-1);
        return ans;
    }
}
